kernel/bpf/verifier.c

```diff
@@ -6076,28 +6077,27 @@ static int check_helper_call(struct bpf_verifier_env *env, struct bpf_insn *insn
 			return -EINVAL;
 		}
 		regs[BPF_REG_0].map_ptr = meta.map_ptr;
-		if (fn->ret_type == RET_PTR_TO_MAP_VALUE) {
+		if (type_may_be_null(ret_type)) {
```

Comment:
Another case of code pulled in that is not present in tag resf_kernel-4.18.0-553.16.1.el8_10.  Please check it.

Reply:
This gets altered in the next commit:
https://github.com/ctrliq/kernel-src-tree/pull/12/commits/82ce5981d565d62536d4f96a41e593e5822a5ac5

```
	regs[BPF_REG_0].type = PTR_TO_MAP_VALUE | ret_flag;
		if (!type_may_be_null(ret_type) &&
		    map_value_has_spin_lock(meta.map_ptr)) {
			regs[BPF_REG_0].id = ++env->id_gen;
```